### PR TITLE
Simplify freddy interface

### PR DESCRIPTION
**Pull request by indrekj**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -48,13 +38,13 @@ freddy.respondTo(destination, callback)
 * `respondTo` returns a promise which resolved with the ResponderHandler
 
 ```coffee
-freddy.respondTo(destination, callback)
+freddy.respondTo destination, (message, handler) ->
+  handler.ack(x: 'y')
```
Comment (urmastalimaa) [inline README.md]: success

File: lib/freddy/freddy_facade.coffee [hunk 2]
```diff
@@ -1,23 +1,33 @@
...
-    @deliver = @producer.produce
     @respondTo = @request.respondTo
     @tapInto = @consumer.tapInto
-    @deliverWithAckAndOptions = @request.deliverWithAckAndOptions
-    @deliverWithResponseAndOptions = @request.deliverWithResponseAndOptions
 
-  shutdown: ->
-    q(@onShutdown())
+  deliver: (destination, message, options, positiveCallback, negativeCallback) ->
+    if typeof options is 'function'
```
Comment (urmastalimaa) [inline lib/freddy/freddy_facade.coffee]: Isn't this equivalent to adding `options = {}` to the signature?

Comment (indrekj) [inline lib/freddy/freddy_facade.coffee]: this is needed to support
`freddy.deliver 'Q', 'data', timeout: 3, pCb, nCb`
`freddy.deliver 'Q', 'data', pCb, nCb`

but yeah, pretty much. the idea is to make the `options` as optional.

File: lib/freddy/freddy_facade.coffee [hunk 3]
```diff
@@ -1,23 +1,33 @@
...
+      negativeCallback = positiveCallback
+      positiveCallback = options
+      options = {}
+
+    if positiveCallback && typeof negativeCallback != 'function'
```
Comment (urmastalimaa) [inline lib/freddy/freddy_facade.coffee]: Maybe also check that `positiveCallback && typeof positiveCallback !='function'`.

Comment (indrekj) [inline lib/freddy/freddy_facade.coffee]: good point. I should also rename these as `successCallback` and `errorCallback`

File: lib/freddy/freddy_facade.coffee [hunk 4]
```diff
@@ -1,23 +1,33 @@
...
+      throw Error('negative callback is required')
+
+    options ||= {}
```
Comment (urmastalimaa) [inline lib/freddy/freddy_facade.coffee]:
```js
options = _.extend({deleteOnTimeout: true, timeout: 3}, _.pick(options, 'timeout', 'suppressLog', 'deleteOnTimeout'))
```

File: lib/freddy/freddy_facade.coffee (hunk 2 above)
Comment (urmastalimaa) [inline lib/freddy/freddy_facade.coffee]: Yeah I see why the default argument doesn't work here.

File: lib/freddy/request.coffee [hunk 5]
```diff
@@ -22,18 +24,16 @@ class Request
...
-    options.timeout ||= DEFAULT_TIMEOUT
-    @_request destination, message, options, (message, msgHandler) =>
-      callback message, msgHandler if (typeof callback is 'function')
+  deliver: (destination, message, options, positiveCallback, negativeCallback) =>
+    if positiveCallback
+      options.type = MESSAGE_TYPES.REQUEST
+      @_request destination, message, options, (message, msgHandler) =>
+        if msgHandler.properties.type == MESSAGE_TYPES.SUCCESS
+          positiveCallback message
+        else if negativeCallback
```
Comment (urmastalimaa) [inline lib/freddy/request.coffee]: Isn't negative callback mandatory?

Comment (urmastalimaa) [inline lib/freddy/request.coffee]: aren't they both necessary in request?

Comment (indrekj) [inline lib/freddy/request.coffee]: forgot to update. you're correct

File: package.json [hunk 6]
```diff
@@ -3,25 +3,28 @@
   "version": "0.2.11",
```
Comment (urmastalimaa) [inline package.json]: update version

**Comment by indrekj** [comment]
Updated

**Pull request merged**